### ISSUE TITLE
fix: publish triggers two jobs that attempt to publish to main

### ISSUE THREAD

**Issue by vernak2539** [vernak2539/astro-rehype-relative-markdown-links#69]
The NPM publish GH Action triggers two actions after publish:

- update changelog
- update docs

Both of these are triggered at the same time, which means the first one to complete will push changes to main, causing the other to break. 

I need to fix this

**Comment by techfg** [comment]
@vernak2539 -

I haven't looked in to this closely but my guess is that because the current workflow uses `gh workflow run` it dispatches the `workflow_dispatch` event resulting in the workflow running asynchronously - just guessing here.

Potentially an alternative would be using either [reusable workflows](https://docs.github.com/en/actions/sharing-automations/reusing-workflows) or [composite actions](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action) which I'm guessing will be synchronous calls (see [avoiding duplications](https://docs.github.com/en/actions/sharing-automations/avoiding-duplication)).

I think we discussed this a while back but my $0.02 is that having the docs and change log built & committed after the package is published isn't really ideal IMHO.  I'm speculating here but I think you are manually updating the version in `package.json`, committing, then tagging which will trigger the workflows.  Assuming this is correct, a couple of ideas:

1. In addition to manually updating `package.json`, you could build docs & changelog locally, commit everything and then let workflows publish
2. Have the workflows build docs, update changelog, commit and then publish (intsead of docs & changelog after).  This would also allow eliminating the manual deletion of the `docs` directory and including docs in the published package (not necessary you may still want the docs not to be included).

Just some food for thought when you circle back to this :)

**Comment by techfg** [comment]
Makes sense about doing changelog & docs after re: potential publish failures.  Only thing I'd recommend then is to remove CHANGELOG from files included in the distributed package since it won't be accurate but I see you already did that in #71 :)

Sorry, what do you mean by `np`?
